src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetDeclaredSymbolAPITests.cs

```diff
@@ -5267,6 +5267,380 @@ void F(int x = 2, = 3) { }
             Assert.Same(symbol1.ContainingSymbol, symbol2.ContainingSymbol);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74348")]
+        public void ObjectInitializerIncompletePropertyValueDeclaration01()
+        {
+            var source = """
+                public class Thing
+                {
+                    public int Key { get; set; }
+                    public string? Value { get; set; }
+                }
+
+                public class Using
+                {
+                    public static Thing CreateThing(int key, string? value)
+                    {
+                        return new()
+                        {
+                            Key = key,
+                            Value,
+                        };
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var root = tree.GetCompilationUnitRoot();
+            var propertyDecls = root.DescendantNodes()
+                .OfType<ClassDeclarationSyntax>()
+                .First()
+                .ChildNodes()
+                .OfType<PropertyDeclarationSyntax>()
+                .ToArray();
+            var valuePropertyDecl = propertyDecls[1];
+            var valueProperty = model.GetDeclaredSymbol(valuePropertyDecl);
+            Assert.NotNull(valueProperty);
+
+            var initializers = root.DescendantNodes()
+                .OfType<InitializerExpressionSyntax>()
+                .First()
+                .ChildNodes()
+                .ToArray();
+            var initializedSymbol = model.GetSymbolInfo(initializers[1]).Symbol;
+
+            Assert.NotNull(initializedSymbol);
```

Comment:
>Assert.NotNull(initializedSymbol); [](http://example.com/codeflow?start=12&length=34)

I think this assert together with the one below can be replaced with 
```
Assert.Equal("...", initializedSymbol.ToTestDisplayString());
```

Reply:
This suggestion is applicable to other added tests too

Reply:
Done